packages/shared/src/hooks/useRefreshToken.ts

```diff
@@ -11,15 +11,17 @@ export function useRefreshToken(
     new Date(accessToken?.expiresIn),
     new Date(),
   );
-  const [useRefresh] = useDebounce(refresh, difference - 1000 * 60 * 2);
+  const differencePlusTwoMinutes = difference - 1000 * 60 * 2;
+  const saveDelay =
+    differencePlusTwoMinutes <= 200 ? 200 : differencePlusTwoMinutes;
+
+  const [useRefresh] = useDebounce(refresh, saveDelay, 1000 * 60);
 
   useEffect(() => {
-    if (accessToken) {
+    if (accessToken?.token) {
       // @NOTE see https://dailydotdev.atlassian.net/l/cp/dK9h1zoM
       // eslint-disable-next-line react-hooks/rules-of-hooks
```

Comment:
We don't need this anymore right?

Reply:
We do, it's a hook call inside a hook call (technically not according to React rules 😅)

Reply:
You gotta love these rules

Reply:
if you just rename it to something not starting with `use` we can remove it. `useRefresh` is not really a hook.